### DISCUSSION TITLE
Adding on_drag_* callbacks to pydeck

### DISCUSSION THREAD

**Discussion by danielXr** [visgl/deck.gl#5132] [Q&A]
Hi,

**tl;dr;**
How can I implement callbacks to mouse dragging in pydeck and be able to run it both in Jupyter and as a standalone HTML?
Tried to find documentations or examples and found none.

I'm writing a small tool where I want the ability to move data from one layer in 3D, while the other layers shouldn't move.
The idea is to find the best translation and rotation of a cloud of points (using PointCloudLayer) so that they will fit best on the displayed mesh (using TerrainLayer).
Eventually I would like to save the results in some file.

Thanks,
Daniel.

#### Environment:
 - Framework Version: pydeck 0.5.0
- Jupyter notebook 5.7.10
 - OS: Mac OS X 10.15.7

**Comment by ajduberstein** [comment]
The callbacks that pydeck currently supports are [here in the docs](https://pydeck.gl/event_handling.html)—the ones you’re looking for are not in pydeck yet. You might want to write this as a [custom layer](https://pydeck.gl/gallery/custom_layer.html) which you can then import into pydeck.

**Comment by Pessimistress** [comment]
I suggest you try it with JavaScript first. All layers support [onDrag*](https://deck.gl/docs/api-reference/core/layer#interaction-properties) callbacks in JS. If those are indeed what you need, pydeck can expose them in the Python API.

**Comment by danielXr** [comment]
@Pessimistress thanks for your reply. This is what I need. Can it be added to pyDeck?

**Comment by danielXr** [comment]
This something I also really need.  When will it be possible to add the drag evens?
* onDragStart
* onDrag
* onDragEnd

**Reply by ajduberstein** [reply]
It's straightforward to add. I'll work on this.

**Reply by danielXr** [reply]
@ajduberstein this will be great, thanks!
When do you believe it will be available?

**Reply by ajduberstein** [reply]
The change is merged–you can now build pydeck now from its source and try it out.

**Reply by danielXr** [reply]
@ajduberstein  thank you, will try it out.
Also, when are you going to release an updated version with this?

**Comment by ajduberstein** [comment]
Addressed in https://github.com/visgl/deck.gl/pull/5963
